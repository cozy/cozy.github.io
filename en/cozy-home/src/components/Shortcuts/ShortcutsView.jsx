import React from 'react'
import cx from 'classnames'

import Divider from 'cozy-ui/transpiled/react/Divider'

import { ShortcutLink } from '@/components/ShortcutLink'

export const ShortcutsView = ({ shortcutsDirectories }) => {
  return !shortcutsDirectories ? null : (
    <>
      {shortcutsDirectories.map(directory => (
        <div
          key={directory.name}
          className="shortcuts-list-wrapper u-m-auto u-w-100"
        >
          <Divider className="u-mv-0" variant="subtitle2">
            {directory.name}
          </Divider>
          <div
            className={cx(
              'shortcuts-list shortcuts-list--gutter u-w-100 u-mh-auto u-mv-3 u-mv-2-t u-flex-justify-center'
            )}
          >
            {directory.items.map(shortcut => (
              <ShortcutLink key={shortcut.name} file={shortcut} />
            ))}
          </div>
        </div>
      ))}
    </>
  )
}
