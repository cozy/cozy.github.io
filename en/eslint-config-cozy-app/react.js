'use strict'

const basics = require('./basics')

module.exports = {
  plugins: basics.plugins.concat(['react-hooks']),
  extends: basics.extends.concat(['plugin:react/recommended']),
  parser: basics.parser,
  parserOptions: { ecmaFeatures: { jsx: true } },
  env: basics.env,
  settings: { react: { version: 'detect' } },
  rules: Object.assign({}, basics.rules, {
    'react/prop-types': 'off',
    'react/jsx-curly-brace-presence': [
      'error',
      { props: 'never', children: 'never' }
    ],
    'react-hooks/rules-of-hooks': 'error',
    'react-hooks/exhaustive-deps': 'warn'
  }),
  overrides: [
    {
      extends: [
        'eslint:recommended',
        'plugin:@typescript-eslint/eslint-recommended',
        'plugin:@typescript-eslint/recommended-type-checked',
        'plugin:@typescript-eslint/strict'
      ],

      files: ['**/*.ts', '**/*.tsx'],
      parser: '@typescript-eslint/parser',
      parserOptions: {
        project: 'packages/**/tsconfig.json',
        tsconfigRootDir: './'
      },
      plugins: ['@typescript-eslint'],
      rules: {
        '@typescript-eslint/explicit-function-return-type': 'error',
        '@typescript-eslint/no-explicit-any': 'error',
        '@typescript-eslint/no-unused-vars': [
          'error',
          { ignoreRestSiblings: true }
        ],
        '@typescript-eslint/no-invalid-void-type': 0,
        '@typescript-eslint/no-non-null-assertion': 0,
        '@typescript-eslint/no-useless-constructor': 1,
        '@typescript-eslint/no-dynamic-delete': 1
      }
    },
    {
      files: ['**/*.spec.jsx', '**/*.spec.js', '**/*.spec.tsx', '**/*.spec.ts'],
      rules: {
        'react/display-name': ['off']
      }
    }
  ]
}
