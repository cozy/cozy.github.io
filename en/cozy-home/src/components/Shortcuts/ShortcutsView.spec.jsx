import React from 'react'

import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import { ShortcutsView } from './ShortcutsView'
import AppLike from '@/test/AppLike'

describe('Shortcuts', () => {
  it('Should display nothing if nothing was found', () => {
    const { container } = render(
      <AppLike>
        <ShortcutsView shortcutsDirectories={null} />
      </AppLike>
    )
    expect(container).toMatchSnapshot()
  })

  it('Should display a shortcut directory with its files', async () => {
    const listName = 'List title'
    const listItem = 'List item'

    const root = render(
      <AppLike>
        <ShortcutsView
          shortcutsDirectories={[
            { name: listName, items: [{ name: listItem }] }
          ]}
        />
      </AppLike>
    )
    expect(root).toMatchSnapshot()
  })

  it('Should display multiple sections for multiple directories', () => {
    const shortcutsDirectories = [
      { name: 'a', items: [{ name: 'b' }] },
      { name: 'c', items: [{ name: 'd' }] }
    ]

    const root = render(
      <AppLike>
        <ShortcutsView shortcutsDirectories={shortcutsDirectories} />
      </AppLike>
    )

    expect(root).toMatchSnapshot()
  })
})
