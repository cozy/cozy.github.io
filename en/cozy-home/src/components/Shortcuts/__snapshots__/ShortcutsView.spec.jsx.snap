// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shortcuts Should display a shortcut directory with its files 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="TwakeTheme--light-normal u-dc"
      >
        <div
          class="shortcuts-list-wrapper u-m-auto u-w-100"
        >
          <h6
            class="MuiTypography-root styles__divider___SuA5q u-mv-0 MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
          >
            List title
          </h6>
          <div
            class="shortcuts-list shortcuts-list--gutter u-w-100 u-mh-auto u-mv-3 u-mv-2-t u-flex-justify-center"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineNone scale-hover MuiTypography-colorPrimary"
              href="#"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="makeStyles-tileWrapper-6"
                data-testid="square-app-icon"
              >
                <div
                  class="TwakeTheme--light-normal u-dc"
                >
                  <span
                    class="MuiBadge-root-9"
                  >
                    <span
                      class="MuiBadge-root-9 styles__SquareAppIcon-wrapper___2SEuM styles__SquareAppIcon-wrapper--light___2iR5Y undefined"
                    >
                      <h2
                        class="MuiTypography-root-32 makeStyles-letter-4 MuiTypography-h2-38 MuiTypography-colorTextPrimary-57 MuiTypography-alignCenter-48"
                      >
                        L
                      </h2>
                      <span
                        class="MuiBadge-badge-10 badgeSizeLarge MuiBadge-anchorOriginTopRightRectangular-16"
                      />
                    </span>
                    <span
                      class="MuiBadge-badge-10 makeStyles-qualifier-8 MuiBadge-anchorOriginBottomRightRectangular-18"
                    >
                      <svg
                        class="styles__icon___23x3R"
                        height="10"
                        viewBox="0 0 16 16"
                        width="10"
                      >
                        <path
                          d="M14.222 14.222H1.778V1.778H8V0H1.778C.79 0 0 .8 0 1.778v12.444C0 15.2.791 16 1.778 16h12.444C15.2 16 16 15.2 16 14.222V8h-1.778v6.222zM9.778 0v1.778h3.19l-8.737 8.738 1.253 1.253 8.738-8.738v3.191H16V0H9.778z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
                <h6
                  class="MuiTypography-root makeStyles-name-1 u-spacellipsis MuiTypography-h6 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
                >
                  List item
                </h6>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="TwakeTheme--light-normal u-dc"
    >
      <div
        class="shortcuts-list-wrapper u-m-auto u-w-100"
      >
        <h6
          class="MuiTypography-root styles__divider___SuA5q u-mv-0 MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
        >
          List title
        </h6>
        <div
          class="shortcuts-list shortcuts-list--gutter u-w-100 u-mh-auto u-mv-3 u-mv-2-t u-flex-justify-center"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineNone scale-hover MuiTypography-colorPrimary"
            href="#"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="makeStyles-tileWrapper-6"
              data-testid="square-app-icon"
            >
              <div
                class="TwakeTheme--light-normal u-dc"
              >
                <span
                  class="MuiBadge-root-9"
                >
                  <span
                    class="MuiBadge-root-9 styles__SquareAppIcon-wrapper___2SEuM styles__SquareAppIcon-wrapper--light___2iR5Y undefined"
                  >
                    <h2
                      class="MuiTypography-root-32 makeStyles-letter-4 MuiTypography-h2-38 MuiTypography-colorTextPrimary-57 MuiTypography-alignCenter-48"
                    >
                      L
                    </h2>
                    <span
                      class="MuiBadge-badge-10 badgeSizeLarge MuiBadge-anchorOriginTopRightRectangular-16"
                    />
                  </span>
                  <span
                    class="MuiBadge-badge-10 makeStyles-qualifier-8 MuiBadge-anchorOriginBottomRightRectangular-18"
                  >
                    <svg
                      class="styles__icon___23x3R"
                      height="10"
                      viewBox="0 0 16 16"
                      width="10"
                    >
                      <path
                        d="M14.222 14.222H1.778V1.778H8V0H1.778C.79 0 0 .8 0 1.778v12.444C0 15.2.791 16 1.778 16h12.444C15.2 16 16 15.2 16 14.222V8h-1.778v6.222zM9.778 0v1.778h3.19l-8.737 8.738 1.253 1.253 8.738-8.738v3.191H16V0H9.778z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
              <h6
                class="MuiTypography-root makeStyles-name-1 u-spacellipsis MuiTypography-h6 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
              >
                List item
              </h6>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Shortcuts Should display multiple sections for multiple directories 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="TwakeTheme--light-normal u-dc"
      >
        <div
          class="shortcuts-list-wrapper u-m-auto u-w-100"
        >
          <h6
            class="MuiTypography-root styles__divider___SuA5q u-mv-0 MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
          >
            a
          </h6>
          <div
            class="shortcuts-list shortcuts-list--gutter u-w-100 u-mh-auto u-mv-3 u-mv-2-t u-flex-justify-center"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineNone scale-hover MuiTypography-colorPrimary"
              href="#"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="makeStyles-tileWrapper-67"
                data-testid="square-app-icon"
              >
                <div
                  class="TwakeTheme--light-normal u-dc"
                >
                  <span
                    class="MuiBadge-root-70"
                  >
                    <span
                      class="MuiBadge-root-70 styles__SquareAppIcon-wrapper___2SEuM styles__SquareAppIcon-wrapper--light___2iR5Y undefined"
                    >
                      <h2
                        class="MuiTypography-root-93 makeStyles-letter-65 MuiTypography-h2-99 MuiTypography-colorTextPrimary-118 MuiTypography-alignCenter-109"
                      >
                        B
                      </h2>
                      <span
                        class="MuiBadge-badge-71 badgeSizeLarge MuiBadge-anchorOriginTopRightRectangular-77"
                      />
                    </span>
                    <span
                      class="MuiBadge-badge-71 makeStyles-qualifier-69 MuiBadge-anchorOriginBottomRightRectangular-79"
                    >
                      <svg
                        class="styles__icon___23x3R"
                        height="10"
                        viewBox="0 0 16 16"
                        width="10"
                      >
                        <path
                          d="M14.222 14.222H1.778V1.778H8V0H1.778C.79 0 0 .8 0 1.778v12.444C0 15.2.791 16 1.778 16h12.444C15.2 16 16 15.2 16 14.222V8h-1.778v6.222zM9.778 0v1.778h3.19l-8.737 8.738 1.253 1.253 8.738-8.738v3.191H16V0H9.778z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
                <h6
                  class="MuiTypography-root makeStyles-name-62 u-spacellipsis MuiTypography-h6 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
                >
                  b
                </h6>
              </div>
            </a>
          </div>
        </div>
        <div
          class="shortcuts-list-wrapper u-m-auto u-w-100"
        >
          <h6
            class="MuiTypography-root styles__divider___SuA5q u-mv-0 MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
          >
            c
          </h6>
          <div
            class="shortcuts-list shortcuts-list--gutter u-w-100 u-mh-auto u-mv-3 u-mv-2-t u-flex-justify-center"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineNone scale-hover MuiTypography-colorPrimary"
              href="#"
              rel="noopener noreferrer"
              target="_blank"
            >
              <div
                class="makeStyles-tileWrapper-67"
                data-testid="square-app-icon"
              >
                <div
                  class="TwakeTheme--light-normal u-dc"
                >
                  <span
                    class="MuiBadge-root-124"
                  >
                    <span
                      class="MuiBadge-root-124 styles__SquareAppIcon-wrapper___2SEuM styles__SquareAppIcon-wrapper--light___2iR5Y undefined"
                    >
                      <h2
                        class="MuiTypography-root-147 makeStyles-letter-65 MuiTypography-h2-153 MuiTypography-colorTextPrimary-172 MuiTypography-alignCenter-163"
                      >
                        D
                      </h2>
                      <span
                        class="MuiBadge-badge-125 badgeSizeLarge MuiBadge-anchorOriginTopRightRectangular-131"
                      />
                    </span>
                    <span
                      class="MuiBadge-badge-125 makeStyles-qualifier-123 MuiBadge-anchorOriginBottomRightRectangular-133"
                    >
                      <svg
                        class="styles__icon___23x3R"
                        height="10"
                        viewBox="0 0 16 16"
                        width="10"
                      >
                        <path
                          d="M14.222 14.222H1.778V1.778H8V0H1.778C.79 0 0 .8 0 1.778v12.444C0 15.2.791 16 1.778 16h12.444C15.2 16 16 15.2 16 14.222V8h-1.778v6.222zM9.778 0v1.778h3.19l-8.737 8.738 1.253 1.253 8.738-8.738v3.191H16V0H9.778z"
                        />
                      </svg>
                    </span>
                  </span>
                </div>
                <h6
                  class="MuiTypography-root makeStyles-name-62 u-spacellipsis MuiTypography-h6 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
                >
                  d
                </h6>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="TwakeTheme--light-normal u-dc"
    >
      <div
        class="shortcuts-list-wrapper u-m-auto u-w-100"
      >
        <h6
          class="MuiTypography-root styles__divider___SuA5q u-mv-0 MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
        >
          a
        </h6>
        <div
          class="shortcuts-list shortcuts-list--gutter u-w-100 u-mh-auto u-mv-3 u-mv-2-t u-flex-justify-center"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineNone scale-hover MuiTypography-colorPrimary"
            href="#"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="makeStyles-tileWrapper-67"
              data-testid="square-app-icon"
            >
              <div
                class="TwakeTheme--light-normal u-dc"
              >
                <span
                  class="MuiBadge-root-70"
                >
                  <span
                    class="MuiBadge-root-70 styles__SquareAppIcon-wrapper___2SEuM styles__SquareAppIcon-wrapper--light___2iR5Y undefined"
                  >
                    <h2
                      class="MuiTypography-root-93 makeStyles-letter-65 MuiTypography-h2-99 MuiTypography-colorTextPrimary-118 MuiTypography-alignCenter-109"
                    >
                      B
                    </h2>
                    <span
                      class="MuiBadge-badge-71 badgeSizeLarge MuiBadge-anchorOriginTopRightRectangular-77"
                    />
                  </span>
                  <span
                    class="MuiBadge-badge-71 makeStyles-qualifier-69 MuiBadge-anchorOriginBottomRightRectangular-79"
                  >
                    <svg
                      class="styles__icon___23x3R"
                      height="10"
                      viewBox="0 0 16 16"
                      width="10"
                    >
                      <path
                        d="M14.222 14.222H1.778V1.778H8V0H1.778C.79 0 0 .8 0 1.778v12.444C0 15.2.791 16 1.778 16h12.444C15.2 16 16 15.2 16 14.222V8h-1.778v6.222zM9.778 0v1.778h3.19l-8.737 8.738 1.253 1.253 8.738-8.738v3.191H16V0H9.778z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
              <h6
                class="MuiTypography-root makeStyles-name-62 u-spacellipsis MuiTypography-h6 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
              >
                b
              </h6>
            </div>
          </a>
        </div>
      </div>
      <div
        class="shortcuts-list-wrapper u-m-auto u-w-100"
      >
        <h6
          class="MuiTypography-root styles__divider___SuA5q u-mv-0 MuiTypography-subtitle2 MuiTypography-colorTextPrimary"
        >
          c
        </h6>
        <div
          class="shortcuts-list shortcuts-list--gutter u-w-100 u-mh-auto u-mv-3 u-mv-2-t u-flex-justify-center"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineNone scale-hover MuiTypography-colorPrimary"
            href="#"
            rel="noopener noreferrer"
            target="_blank"
          >
            <div
              class="makeStyles-tileWrapper-67"
              data-testid="square-app-icon"
            >
              <div
                class="TwakeTheme--light-normal u-dc"
              >
                <span
                  class="MuiBadge-root-124"
                >
                  <span
                    class="MuiBadge-root-124 styles__SquareAppIcon-wrapper___2SEuM styles__SquareAppIcon-wrapper--light___2iR5Y undefined"
                  >
                    <h2
                      class="MuiTypography-root-147 makeStyles-letter-65 MuiTypography-h2-153 MuiTypography-colorTextPrimary-172 MuiTypography-alignCenter-163"
                    >
                      D
                    </h2>
                    <span
                      class="MuiBadge-badge-125 badgeSizeLarge MuiBadge-anchorOriginTopRightRectangular-131"
                    />
                  </span>
                  <span
                    class="MuiBadge-badge-125 makeStyles-qualifier-123 MuiBadge-anchorOriginBottomRightRectangular-133"
                  >
                    <svg
                      class="styles__icon___23x3R"
                      height="10"
                      viewBox="0 0 16 16"
                      width="10"
                    >
                      <path
                        d="M14.222 14.222H1.778V1.778H8V0H1.778C.79 0 0 .8 0 1.778v12.444C0 15.2.791 16 1.778 16h12.444C15.2 16 16 15.2 16 14.222V8h-1.778v6.222zM9.778 0v1.778h3.19l-8.737 8.738 1.253 1.253 8.738-8.738v3.191H16V0H9.778z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
              <h6
                class="MuiTypography-root makeStyles-name-62 u-spacellipsis MuiTypography-h6 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
              >
                d
              </h6>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Shortcuts Should display nothing if nothing was found 1`] = `
<div>
  <div
    class="TwakeTheme--light-normal u-dc"
  />
</div>
`;
